app/models/file_upload.rb

```diff
@@ -0,0 +1,7 @@
+class FileUpload < ApplicationRecord
```

Comment:
I can see this replacing `processed_files` and `temporary_files` using ActiveStorage instead of serializing the file contents.